src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
We should definitely continue the conversation when we're all in the office. In the case where you are using these APIs as a program of any kind which is self-hosting an http endpoint it's quite likely that automatically picking up the "hosting.json" and ASPNET_ env vars is actually the wrong thing to do. Too much detail to go into why that's the case on a github thread.

Also - the problems you're bumping into about how you have to call Build before being able to change Addresses is a sign that ServerFeatures interface is the wrong way to control that. You should be able to provide the addresses on the Builder.